<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Prof Peter Kille">
<meta name="dcterms.date" content="2025-10-16">

<title>Gene Expression Omnibus (GEO)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<style>

      .quarto-title-block .quarto-title-banner h1,
      .quarto-title-block .quarto-title-banner h2,
      .quarto-title-block .quarto-title-banner h3,
      .quarto-title-block .quarto-title-banner h4,
      .quarto-title-block .quarto-title-banner h5,
      .quarto-title-block .quarto-title-banner h6
      {
        color: white;
      }

      .quarto-title-block .quarto-title-banner {
        color: white;
background: darkred;
      }
</style>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./5.0_Transcriptomics_Introduction.html">Session 5. Transcriptomics</a></li><li class="breadcrumb-item"><a href="./5.2_GEO_Open_Data_Archive.html">Gene Expression Omnibus (GEO)</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Gene Expression Omnibus (GEO)</h1>
            <p class="subtitle lead">Open Data Archive for Transcriptomic Data</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Prof Peter Kille </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">October 16, 2025</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Welcome to Bioinformatics!</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction to Big Data Biology and Bioinformatics</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Session 0. HPC and Personal Cloud</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./0.1_Personal_Cloud.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Personal Cloud</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Session 1. Linux: The basics</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./1.1_Introduction_to_Linux.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction to Linux</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./1.2_Loading_Software_Accessing_Data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Loading Software and Accessing Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./1.3_Commandline_Tools_and_Scripting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Extension: Commandline Tools and Scripting</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
 <span class="menu-text">Session 2. NGS data</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./2.1_NGS_Quality_Control.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">NGS Quality Control</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./2.2_genome_assembly.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Genome Assembly</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./2.3_hybrid_assemblies.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Extension: Hybrid Assemblies</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true">
 <span class="menu-text">Session 3. Genomics</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./3.1_Genome_visualisation_and_annotation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Genome Visualisation and Annotation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./3.2_Annotating_mitochondria.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Extension: Annotating Mitochondrial Genomes</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="true">
 <span class="menu-text">Session 4. Building phylogenies</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./4.1_phylogentics_phylogenomics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Phylogenetics and Phylogeneomics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./4.2_phylogentics_phylogenomics_extension.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Extension: Phylogenetics and Phylogeneomics</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="true">
 <span class="menu-text">Session 5. Transcriptomics</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-7" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./5.0_Transcriptomics_Introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Transcriptomics: An Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./5.1_RNAseq_processing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RNAseq Processing</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./5.2_GEO_Open_Data_Archive.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Gene Expression Omnibus (GEO)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./5.3_DEG_Generation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Data Processing : Generation of DEG Lists</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./5.4_DEG_Functional_Interpretation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Functional Interpretation of DEG Lists</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./5.5_Refined_Data_Representations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Refined Data Representation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" aria-expanded="true">
 <span class="menu-text">Session 6. Microbial community analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-8" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./6.1_metabarcoding.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Metabarcoding</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#mining-gene-expression-omnibus-geo" id="toc-mining-gene-expression-omnibus-geo" class="nav-link active" data-scroll-target="#mining-gene-expression-omnibus-geo">Mining Gene Expression Omnibus (GEO)</a></li>
  <li><a href="#a-sra---fastq-and-annotation-serise-matrix-files" id="toc-a-sra---fastq-and-annotation-serise-matrix-files" class="nav-link" data-scroll-target="#a-sra---fastq-and-annotation-serise-matrix-files">2.1A SRA - Fastq and Annotation (Serise Matrix Files)</a>
  <ul class="collapse">
  <li><a href="#sra-ncbi-download" id="toc-sra-ncbi-download" class="nav-link" data-scroll-target="#sra-ncbi-download">SRA NCBI download</a></li>
  <li><a href="#ena-as-an-alternative-portal-to-sra" id="toc-ena-as-an-alternative-portal-to-sra" class="nav-link" data-scroll-target="#ena-as-an-alternative-portal-to-sra">ENA as an alternative portal to SRA</a></li>
  <li><a href="#extracting-metadata" id="toc-extracting-metadata" class="nav-link" data-scroll-target="#extracting-metadata">Extracting metadata</a></li>
  </ul></li>
  <li><a href="#b-raw-count-matrix-csvtsv-and-annotation-serise-matrix-files" id="toc-b-raw-count-matrix-csvtsv-and-annotation-serise-matrix-files" class="nav-link" data-scroll-target="#b-raw-count-matrix-csvtsv-and-annotation-serise-matrix-files">2.1B Raw Count matrix (CSV/TSV) and Annotation (Serise Matrix Files)</a></li>
  <li><a href="#c-geo2r-configured-unprocessed-raw-counts-and-group-metadata" id="toc-c-geo2r-configured-unprocessed-raw-counts-and-group-metadata" class="nav-link" data-scroll-target="#c-geo2r-configured-unprocessed-raw-counts-and-group-metadata">2.1C Geo2R configured unprocessed raw counts and group metadata</a>
  <ul class="collapse">
  <li><a href="#identify-a-geo2r-compatible-dataset" id="toc-identify-a-geo2r-compatible-dataset" class="nav-link" data-scroll-target="#identify-a-geo2r-compatible-dataset">1. Identify a Geo2R compatible dataset</a></li>
  <li><a href="#customise-the-option-and-analyse" id="toc-customise-the-option-and-analyse" class="nav-link" data-scroll-target="#customise-the-option-and-analyse">2. Customise the Option and Analyse</a></li>
  <li><a href="#generate-a-deg-list" id="toc-generate-a-deg-list" class="nav-link" data-scroll-target="#generate-a-deg-list">3. Generate a DEG list</a></li>
  </ul></li>
  <li><a href="#d-processed-dataset" id="toc-d-processed-dataset" class="nav-link" data-scroll-target="#d-processed-dataset">2.1D Processed dataset</a>
  <ul class="collapse">
  <li><a href="#search-for-data-of-intereste" id="toc-search-for-data-of-intereste" class="nav-link" data-scroll-target="#search-for-data-of-intereste">1. Search for data of intereste</a></li>
  <li><a href="#refine-search-to-only-show-datasets-and-series" id="toc-refine-search-to-only-show-datasets-and-series" class="nav-link" data-scroll-target="#refine-search-to-only-show-datasets-and-series">2. Refine search to only show ‘DataSets’ and ‘Series’</a></li>
  <li><a href="#select-procesed-dataset-should-be-number-3-in-list-and-have-a-heat-map-icon-to-the-right" id="toc-select-procesed-dataset-should-be-number-3-in-list-and-have-a-heat-map-icon-to-the-right" class="nav-link" data-scroll-target="#select-procesed-dataset-should-be-number-3-in-list-and-have-a-heat-map-icon-to-the-right">3. Select procesed dataset (should be number 3 in list and have a heat map icon to the right)</a></li>
  <li><a href="#compare-experiment-samples" id="toc-compare-experiment-samples" class="nav-link" data-scroll-target="#compare-experiment-samples">4. Compare experiment samples</a></li>
  <li><a href="#download-degs" id="toc-download-degs" class="nav-link" data-scroll-target="#download-degs">5. Download DEGs</a></li>
  <li><a href="#open-and-merge-deg-lists-in-excel" id="toc-open-and-merge-deg-lists-in-excel" class="nav-link" data-scroll-target="#open-and-merge-deg-lists-in-excel">6. Open and Merge DEG lists in Excel</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<section id="mining-gene-expression-omnibus-geo" class="level1">
<h1>Mining Gene Expression Omnibus (GEO)</h1>
<p>All research articles published in peer-reviewed journals must submit the data (raw or processed) to an accessible data archive - the most commonly used being NCBI GEO. Unfortunately the standards and types of data acceptable for submission to GEO have changed over the years from microarray datasets that were uploaded fully processed and ready to analyse - <a href="#2.1D%20Processed%20dataset">Processed Datasets</a> to current practices where links are provided to the raw fastq file deposited within the Short Read Archive (SRA). Below I have provided instructions of how to create gene lists from each data type provided by GEO (I am sure I would have missed one but hopefully the list cover the majority of the options).</p>
</section>
<section id="a-sra---fastq-and-annotation-serise-matrix-files" class="level1">
<h1>2.1A SRA - Fastq and Annotation (Serise Matrix Files)</h1>
<p>Provision of SRA and annotation files in teh forms of a ‘Serise Matrix Files’ requires you to download the data and create count file by mapping as described in [tutorial 5.1] (https://peter-kille.github.io/Bioinformatics_TrainingWeb/5.1_RNAseq_processing.html). Additionally you will need to reconfigure the annotation provided in the Serise Matrix Files into a metadata file format that can be used for SARTools (DESeq2) pipeline for DEG generation <a href="https://peter-kille.github.io/Bioinformatics_TrainingWeb/5.3_DEG_Generation.html">tutorial 5.3</a>.</p>
<section id="sra-ncbi-download" class="level2">
<h2 class="anchored" data-anchor-id="sra-ncbi-download">SRA NCBI download</h2>
<p>Samples preconfigured for SRA download ofen have <code>SRA Run Selector</code> as a link at the bottom of the entry. Approaches to downloading SRA data and annotation is documented in <a href="https://www.ncbi.nlm.nih.gov/sra/docs/sradownload/#download-sequence-data-from-the-">NCBI Guide</a> and there is a relatively straightforward read <a href="https://www.ncbi.nlm.nih.gov/Traces/index.html?view=run_browser">achive website</a> to make this very easy through a browser interface.</p>
</section>
<section id="ena-as-an-alternative-portal-to-sra" class="level2">
<h2 class="anchored" data-anchor-id="ena-as-an-alternative-portal-to-sra">ENA as an alternative portal to SRA</h2>
<p>The <a href="https://www.ebi.ac.uk/ena/browser/home">European Nucleotide Archive (ENA)</a> provides a mirror of the SRA and sometimes both teh serach functions and the download can be easier. If you have a list of accession numbers you can calculate the download addresses for the accession using a simple <a href="https://www.ebi.ac.uk/ena/browser/downloading-data">FTP guide</a>.</p>
</section>
<section id="extracting-metadata" class="level2">
<h2 class="anchored" data-anchor-id="extracting-metadata">Extracting metadata</h2>
<p>Metadata is contained with the ‘Serise Matrix Files’ - key annotation parameters are usually found under the ‘!Sample_characteristics_ch1’ or ‘!Sample_treatment_protocol_ch1’ with the SRA code under ‘!Sample_relation’. However, both the NCBI and SRA dowbload tools provide the ability to download annotations as a TSV - but sadly sometimes you have to go digging into the ‘Serise Matrix Files’ or even the papers !!</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-1-contents" aria-controls="callout-1" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Workshop Example: Raw Data
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-1" class="callout-1-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>This example is used as the data for <a href="https://peter-kille.github.io/Bioinformatics_TrainingWeb/5.1_RNAseq_processing.html">workshop 5.1</a></p>
<p>The data is derived from NCBI Bioproject <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA658134">PRJNA658134</a> Geo dataset <a href="https://www.ncbi.nlm.nih.gov/gds/?term=GSE156507%5BAccession%5D">GSE156507</a>. For workshop 5.1 I have selected only those L3 stage samples and sub-sampled the data from the 60M reads deposited to ~10M reads so that it can be processed with the workshop but if you want you could download and try and use the full dataset</p>
</div>
</div>
</div>
</section>
</section>
<section id="b-raw-count-matrix-csvtsv-and-annotation-serise-matrix-files" class="level1">
<h1>2.1B Raw Count matrix (CSV/TSV) and Annotation (Serise Matrix Files)</h1>
<p>Many upload now contain a count file and the annotation contained within the ‘Serise Matrix Files’. In this case you can extract the metadata from the annotation file as described above - <a href="##%20Extracting%20metadata">## Extracting metadata</a>.</p>
<p>To use the count file TSV matrix file you will need to split it into separate files and save it to a folder called ‘FeatureCounts’. Each file needs to be named by the sample name - usually given as the header for the count column and contain 2 tab deliminated columns the gene ID and the count - no headers. You then need to encorperate these file names into your metadata file for SARTools (DESeq2) and you can run generate DEGs as descrived in <a href="https://peter-kille.github.io/Bioinformatics_TrainingWeb/5.3_DEG_Generation.html">tutorial 5.3</a>.</p>
</section>
<section id="c-geo2r-configured-unprocessed-raw-counts-and-group-metadata" class="level1">
<h1>2.1C Geo2R configured unprocessed raw counts and group metadata</h1>
<p>Some data has been uploaded into GEO in a form that allows it to be analysed with a web-based R script based on DESeq2 - called Geo2R. The metadata for these data sets have been extracted and can be used to ‘define’ groups for subsequent DEG generation.</p>
<section id="identify-a-geo2r-compatible-dataset" class="level2">
<h2 class="anchored" data-anchor-id="identify-a-geo2r-compatible-dataset">1. Identify a Geo2R compatible dataset</h2>
<p>Select un-processed series (should be number 2 - it will have icon <strong><em>&lt; Analyze with GEO2R &gt;</em></strong> at the end of the entry)</p>
<p>Click on <strong><em>&lt; Analyze with GEO2R &gt;</em></strong></p>
<p>Now Define groups by click clicking the <strong><em>&lt; Define Group pull &gt;</em></strong> down and create groups ‘control’ and ‘Treatment’ (enter group name and press enter). Click on each sample in list and associated it with one of your group (you can hold ctrl down to select multiple entry before associating them with the group).</p>
<p><img src="images/compare_groups_geo2R.png" class="img-fluid" alt="Select Groups to Compare"> <strong><em>Select Groups to compare</em></strong></p>
</section>
<section id="customise-the-option-and-analyse" class="level2">
<h2 class="anchored" data-anchor-id="customise-the-option-and-analyse">2. Customise the Option and Analyse</h2>
<p>Select <strong><em>&lt; Options &gt;</em></strong> and customise as shown below:</p>
<p><img src="images/Geo2R_Options.png" class="img-fluid" alt="Geo2R Options"> <strong><em>Geo2R Options</em></strong></p>
<p>Select <strong><em>&lt; reanalyze &gt;</em></strong></p>
<p>Will will now see a Processing icon - this may take a minute or two.</p>
</section>
<section id="generate-a-deg-list" class="level2">
<h2 class="anchored" data-anchor-id="generate-a-deg-list">3. Generate a DEG list</h2>
<p>You will now see a table and a series of Visualisations - review the visualisation taking note of what each are showing you.</p>
<p><img src="images/GEO2R_results.png" class="img-fluid" alt="Geo2R results"> <strong><em>Geo2R results</em></strong></p>
<p>Venn diagram showing GSE4567: Limma, Padj&lt;0.05 - 704 genes - this is the set we will use</p>
<p>Click on <strong><em>&lt; Explore and download, control vs treatment and Download Significant genes &gt;</em></strong></p>
<p>This will give you a Tsv you can open in excel</p>
<p><a href="https://www.ncbi.nlm.nih.gov/sites/GDSbrowser?acc=GDS2565">Dataset to use - Geo:GDS2565</a></p>
<p><a href="https://cf.sharepoint.com/:b:/t/SystemsBiology/ESRyTPUlb_dJgpHLXUbYk2wBO_SyL9gXhOXOs-BpHFSPsw?e=FXrOLC">Geo Workshop</a></p>
</section>
</section>
<section id="d-processed-dataset" class="level1">
<h1>2.1D Processed dataset</h1>
<p>Original microarray data and early RNAseq data were upload as fully processed datasets allowing easy (albeit simple) webased analysis. You can identify tehse both as they display a heatmap icon but also because they are listed as ‘datasets’ by GEO.</p>
<section id="search-for-data-of-intereste" class="level2">
<h2 class="anchored" data-anchor-id="search-for-data-of-intereste">1. Search for data of intereste</h2>
<p>Go to Geo DataSets: (https://www.ncbi.nlm.nih.gov/gds/)[https://www.ncbi.nlm.nih.gov/gds/]</p>
<p><img src="images/GEO_DataSets.png" class="img-fluid" alt="GEO_DataSet Search"> <strong><em>GEO_DataSet Search</em></strong></p>
<p>now select <search></search></p>
</section>
<section id="refine-search-to-only-show-datasets-and-series" class="level2">
<h2 class="anchored" data-anchor-id="refine-search-to-only-show-datasets-and-series">2. Refine search to only show ‘DataSets’ and ‘Series’</h2>
<p>Select <strong><em>&lt; DataSets and Series &gt;</em></strong> from the left hand menu.</p>
<p><img src="images/DataSet_Selection.png" class="img-fluid" alt="Refine to DataSets"> <strong><em>Refine to DataSets</em></strong></p>
</section>
<section id="select-procesed-dataset-should-be-number-3-in-list-and-have-a-heat-map-icon-to-the-right" class="level2">
<h2 class="anchored" data-anchor-id="select-procesed-dataset-should-be-number-3-in-list-and-have-a-heat-map-icon-to-the-right">3. Select procesed dataset (should be number 3 in list and have a heat map icon to the right)</h2>
<p>Click on title <strong><em>&lt; Endothelial cell response to ultrafine particles &gt;</em></strong> to select DataSet</p>
<p><img src="images/GDS2565.png" class="img-fluid" alt="Entry Page"> <strong><em>Entry Page</em></strong></p>
<p>Take notes off all pertinent information about the experiment, including species and what microarray platform was used. In this case experiment was conducted on rats and analysed on an Affymetrix Human Genome U133 Plus 2.0 Array. Download any publication available. Also, if you are interested you could have a look at the cluster analysis on the right hand site. This will show you how the relationship of the expression profiles from each sample relative to each other. If the experiment was successful, all samples of a certain treatment should cluster together.</p>
</section>
<section id="compare-experiment-samples" class="level2">
<h2 class="anchored" data-anchor-id="compare-experiment-samples">4. Compare experiment samples</h2>
<p>Click <strong><em>&lt;Compare 2 sets of sample&gt;</em></strong></p>
<p>Choose <strong><em>&lt;test e.g.&nbsp;One-tailed t-test (A &gt; B)&gt;</em></strong></p>
<p>Choose <significance level="">_ e.g.&nbsp;0.001</significance></p>
<p>Click on: Step 2: Select which Samples to put in Group A and Group B</p>
<p><img src="images/group_compare.png" class="img-fluid" alt="Select Groups to Comnpare"> <strong><em>Select Groups to compare</em></strong></p>
<p>Choose <strong><em>&lt; Query Group A vs B &gt;</em></strong></p>
<p>You should now see a list of the following DEGS</p>
<p><img src="images/DEGs.png" class="img-fluid" alt="DEG list"> <strong><em>DEG List</em></strong></p>
</section>
<section id="download-degs" class="level2">
<h2 class="anchored" data-anchor-id="download-degs">5. Download DEGs</h2>
<p>You have &gt;100 DEGs but the page only displays the first 20. Before downloading DEGS change the items per page from 20 to 500.</p>
<p><img src="images/Items_per_page.png" class="img-fluid" alt="Items Per Page"> <strong><em>Items Per Page</em></strong></p>
<p>Select <strong><em>&lt; Download profile data &gt;</em></strong></p>
<p>and you will be prompted to save the profiling of the DEGs displayed as default file name <strong><em>&lt;profile_data.txt&gt;</em></strong></p>
<p>Save to appropriate location. If needed Select Page 2…. of the DEGS and repeat the download process.</p>
<p>Repeat this process for <strong><em>&lt;test e.g.&nbsp;One-tailed t-test (B &gt; A) significance level 0.001 &gt;</em></strong>_</p>
</section>
<section id="open-and-merge-deg-lists-in-excel" class="level2">
<h2 class="anchored" data-anchor-id="open-and-merge-deg-lists-in-excel">6. Open and Merge DEG lists in Excel</h2>
<p>The DEG lists show the gene list with there relative expression level (normalised) and annotation for the genes involved (annotation shown in columns BG -&gt;). For Our next steps we will use the Gene symbol that is in Column BH.</p>
<p><a href="https://cardiff.cloud.panopto.eu/Panopto/Pages/Viewer.aspx?id=fc9e75dd-2cad-4c45-9c90-ab6b00bf41e2">GEO Database Panopto</a></p>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>